### PR TITLE
Update training material to be Kedro 0.15.8

### PR DESCRIPTION
**Pull request by 921kiyo**
## Description
Updated some syntax changes.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking “Submit Pull Request”:

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: kedro-training/docs/07_pipelines.md [hunk 1]
```diff
@@ -621,13 +621,13 @@ It is important to emphasise that the Kedro pipeline is runnable only if _all_ f
...
+These `MemoryDataSet`s pass data across nodes during the run, but are automatically deleted after the run finishes, therefore if you want to have an access to those intermediary datasets after the run, you need to define them in `catalog.yml`.
 
-Let's say, we want to persist the `regressor` dataset to local disk. `regressor` is a `LinearRegression` object returned by the `train_model` node. We can't use, for instance, `CSVLocalDataSet` since it only works with `pandas` DataFrames, but we can instead utilise `PickleLocalDataSet` that allows to serialise Python objects and store them in the file system. Now all you have to do is to add the following dataset definition to `conf/base/catalog.yml`:
+Let's say, we want to persist the `regressor` dataset to local disk. `regressor` is a `LinearRegression` object returned by the `train_model` node. We can't use, for instance, `CSVDataSet` since it only works with `pandas` DataFrames, but we can instead utilise `PickleDataSet` that allows to serialise Python objects and store them in the file system. Now all you have to do is to add the following dataset definition to `conf/base/catalog.yml`:
```
Comment (mzjp2) [inline kedro-training/docs/07_pipelines.md:626]: Should we say `CSVDataSet` or `pandas.CSVDataSet` and likewise for `pickle`?

File: kedro-training/docs/08_transformers.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-2. Developing our own transformer for tracking memory consumption 
+2. Developing our own transformer for tracking memory consumption
 
 ## Applying built-in transformers
 
-Transformers are applied at the `DataCatalog` level. To apply the built-in `ProfileTimeTransformer`, you need to: 
+Transformers are applied at the `DataCatalog` level. To apply the built-in `ProfileTimeTransformer`, you need to:
 1. Navigate to `src/kedro_training/run.py`
-2. Override `_create_catalog` method for your `ProjectContext` class using the following:  
+2. Override `_create_catalog` method for your `ProjectContext` class using the following:
```
Comment (mzjp2) [inline kedro-training/docs/08_transformers.md:11]: Can't comment on the below lines, but line 17 of this file has
```python
from kedro.contrib.io.transformers import ProfileTimeTransformer  # new import
```

need to replace `contrib` by `extras`

File: kedro-training/docs/09_versioning.md [hunk 3]
```diff
@@ -8,12 +8,12 @@ Suppose you want to version `master_table`. To enable versioning, simply add a `
...
-  type: CSVLocalDataSet
+  type: pandas.CSVDataSet
   filepath: data/03_primary/master_table.csv
   versioned: true
 ```
 
-The `DataCatalog` will create a versioned `CSVLocalDataSet` called `master_table`. The actual csv file location will look like `data/03_primary/master_table.csv/<version>/master_table.csv`, where the first `/master_table.csv/` is a directory and `<version>` corresponds to a global save version string formatted as `YYYY-MM-DDThh.mm.ss.sssZ`.
+The `DataCatalog` will create a versioned `CSVDataSet` called `master_table`. The actual csv file location will look like `data/03_primary/master_table.csv/<version>/master_table.csv`, where the first `/master_table.csv/` is a directory and `<version>` corresponds to a global save version string formatted as `YYYY-MM-DDThh.mm.ss.sssZ`.
 
 With the similar way, you can version your machine learning model. Enable versioning for `regressor` as follow:
```
Comment (mzjp2) [inline kedro-training/docs/09_versioning.md:18]: Can't comment on the below code again, but line 22 of this file has `PickleLocalDataSet` that needs to be changed to new format

File: kedro-training/docs/06_jupyter-notebook-workflow.md [hunk 4]
```diff
@@ -98,16 +98,16 @@ which will run all the nodes from your default project pipeline in a sequential
...
+|   `to_nodes`    | `Iterable[str]`  | A list of node names which should be used as an end point                                                                                                                                                                                      |
+|  `from_inputs`  | `Iterable[str]`  | A list of dataset names which should be used as a starting point                                                                                                                                                                               |
+| `load_versions` | `Dict[str, str]` | A mapping of a dataset name to a specific dataset version (timestamp) for loading - this applies to the versioned datasets only                                                                                                                |
+| `pipeline_name` |      `str`       | Name of the modular pipeline to run - must be one of those returned by `create_pipelines` function from `src/<package_name>/pipeline.py`                                                                                                       |
```
[the remark this answers is not in the document]
Comment (921kiyo) [reply kedro-training/docs/06_jupyter-notebook-workflow.md]: `extra_params` is an argument to the constructor, whereas this table is a list of arguments for `context.run()`, so I don't think we need to add this here.

[the remark this answers is not in the document]
Comment (mzjp2) [reply kedro-training/docs/06_jupyter-notebook-workflow.md]: Ah yes, I should have checked. My bad! I thought it was part of `run`.

**Pull request merged**